### PR TITLE
feat(experiments): define custom Trends exposure in the modal

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Problem
In the new metrics view, we no longer use the `Goal` component (see screenshot) to view and edit the custom exposure metric for Trends.

image

## Changes
Add the ability to edit the exposure metric from within the Trends metric modal. This is for primary metrics only. Will repeat this for secondary metrics in a follow up.

image

image

## How did you test this code?
👀

### CODE REVIEW

File: frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx [hunk 1]
```diff
@@ -31,70 +35,245 @@ export function PrimaryGoalTrends(): JSX.Element {
...
+                                                    idx === metricIdx
+                                                        ? {
+                                                              ...metric,
+                                                              exposure_query: {
+                                                                  kind: NodeKind.TrendsQuery,
+                                                                  series: [
+                                                                      {
+                                                                          kind: NodeKind.EventsNode,
+                                                                          name: '$pageview',
+                                                                          event: '$pageview',
```
Comment (danielbachhuber) [inline frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx:169]: What's the rationale for using `$pageview` here?

File: frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx [hunk 2]
```diff
@@ -31,70 +35,245 @@ export function PrimaryGoalTrends(): JSX.Element {
...
+                                                                          .format('YYYY-MM-DDTHH:mm'),
+                                                                      date_to: dayjs()
+                                                                          .endOf('d')
+                                                                          .format('YYYY-MM-DDTHH:mm'),
+                                                                      explicitDate: true,
+                                                                  },
+                                                                  trendsFilter: {
+                                                                      display: ChartDisplayType.ActionsLineGraph,
+                                                                  },
+                                                                  filterTestAccounts: true,
```
Comment (danielbachhuber) [inline frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx:185]: Does `dateRange` and `filterTestAccounts` need to be hard-coded? It seems like they should be derived from other configuration.

File: frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx [hunk 3]
```diff
@@ -31,70 +35,245 @@ export function PrimaryGoalTrends(): JSX.Element {
...
+                                        }}
+                                        readOnly
+                                    />
+                                </div>
+                            </>
+                        ),
+                    },
+                    {
+                        key: 'exposure',
+                        label: 'Exposure',
```
Comment (danielbachhuber) [inline frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx:116]: Oops, I didn't realize we accidentally dropped the exposure metric UI for Trends. I'm surprised we haven't gotten any tickets about that.

**Review by danielbachhuber — APPROVED** [review]
Approving so you can land and iterate on Monday, assuming my questions have reasonable answers 😄

File: frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx (hunk 3 above)
Comment (jurajmajerik) [reply frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx:116]: Uhm we didn't? https://us.posthog.com/project/2/experiments/73149

image

File: frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx (hunk 2 above)
Comment (jurajmajerik) [reply frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx:185]: `dateRange` should be okay as is - it's suitable for the preview (last 14 days), and it gets updated in the query runner when calculating results.

`filterTestAccounts` - probably a good idea..

File: frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx (hunk 1 above)
Comment (jurajmajerik) [reply frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx:169]: Need to set some default event here, and `$pageview` is our default everywhere. But maybe `$feature_flag_called` makes more sense.

File: frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx (hunk 3 above)
Comment (danielbachhuber) [reply frontend/src/scenes/experiments/Metrics/PrimaryGoalTrends.tsx:116]: @jurajmajerik You're right. I was looking at a data warehouse trends experiment where we did remove the UI element.

**Pull request merged**